samza-test/src/main/java/org/apache/samza/test/framework/system/TestRunner.java

```diff
@@ -388,4 +342,41 @@ private TaskFactory createTaskFactory() {
     throw new SamzaException(String.format("Not supported task.class %s. task.class has to implement either StreamTask "
         + "or AsyncStreamTask", taskClass.getName()));
   }
+
+  /**
+   * Creates an in memory stream with {@link InMemorySystemFactory} and initializes its partition stream with messages
+   * @param partitonData key of the map represents partitionId and value represents
+   *                 messages in the partition
+   * @param descriptor describes a stream to initialize with the in memory system
+   */
+  private <StreamMessageType> void initializeInMemoryInputStream(InMemoryInputDescriptor descriptor,
+      Map<Integer, Iterable<StreamMessageType>> partitonData) {
+    String systemName = descriptor.getSystemName();
+    String streamName = (String) descriptor.getPhysicalName().orElse(descriptor.getStreamId());
+    if (configs.containsKey(TaskConfig.INPUT_STREAMS())) {
```

Comment:
Why is this necessary? Shouldn't the job already know what inputs it is listening to?

Reply:
backwards compatibility for for low level task api